### PR TITLE
Add compatibility function Compat.LinAlg.checksquare

### LINKED ISSUE #168: chksquare is now called checksquare
https://github.com/JuliaLang/julia/pull/14601

### PR DESCRIPTION
**Pull request by andreasnoack**
Fixes #168

### CODE REVIEW

**Comment by yuyichao** [comment]
Shouldn't this be better handled by a unexported `Compat.checksquare` or `Compat.LinAlg.checksquare` function?

**Comment by andreasnoack** [comment]
Maybe. What are the trade-offs?

**Comment by yuyichao** [comment]
An actual name is much more reliable than rewriting the ast with a macro and should be the preferred way.

**Comment by yuyichao** [comment]
The down side is of course that you can't just remove the `@compat` when dropping support for old julia version and need to replace `Compat` or `Compat.LinAlg` by `Base.LinAlg` instead.

**Comment by andreasnoack** [comment]
I can see that, but it is compelling that you can just use latest syntax + `@compat`. Do you see any specific vulnerabilities with the rewrite function I've written?

**Comment by yuyichao** [comment]
The fact that you need to patch `using` (and missing `import` and `importall`)? In general, I think we only use `@compat` to bridge syntax changes and while there are changes that are tricky to classify, a renaming like this is certainly not one of those.

File: src/Compat.jl [hunk 1]
```diff
@@ -859,4 +859,14 @@ else  # if VERSION < v"0.5.0-dev+1182"
 
 end
 
+module LinAlg
+    if VERSION < v"0.5.0-dev+2022"
+        checksquare(x) = Base.LinAlg.chksquare(x)
```
Comment (yuyichao) [inline src/Compat.jl]: These can be `const checksquare = Base.LinAlg.chksquare` and `const checksquare = Base.LinAlg.checksquare` (or `import Base.LinAlg.checksquare`).

**Comment by andreasnoack** [comment]
@yuyichao Thanks for the comments. I've updated the PR.

**Comment by yuyichao** [comment]
LGTM. =)

**Pull request merged**